addons/mail/static/src/core/web/mail_composer_template_selector.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates>
+    <t t-name="mail.MailComposerTemplateSelector">
+        <Dropdown t-if="props.record.data.can_edit_body" menuClass="'mail-composer-template-dropdown'">
+            <button class="btn btn-light mail-composer-template-dropdown-btn">
+                <i class="fa fa-fw fa-ellipsis-v"/>
+            </button>
+            <t t-set-slot="content">
+                <div class="px-3 pb-1 text-muted">Insert Template</div>
+                <t t-foreach="state.templates" t-as="template" t-key="template_index">
+                    <DropdownItem onSelected="() => this.onLoadTemplate(template)">
+                        <t t-if="template.display_name" t-out="template.display_name"/>
+                        <span t-else="" class="fst-italic">Untitled</span>
```

Comment:
Here, I put a placeholder when the user does not provide any name for the template. I could maybe make the field required and make sure that the template name is unique?

When two templates have the same name, you can not really tell which template you are deleting/overwriting/etc.

Reply:
Well name is not required, so ... we can't do much about it. Template usage is a mess in a real production environment like our own, we need the search capability otherwise you never find the right one.